lib/exprengine.h

```diff
@@ -40,13 +40,18 @@ class Variable;
 #if defined(__GNUC__) && defined (__SIZEOF_INT128__)
 typedef __int128_t int128_t;
 #else
+#if defined(USE_BOOST_MULTIPREC)
```

Comment:
This should be the first case. If we configure to use it we should use it across all platforms